letsencrypt/auth_handler.py

```diff
@@ -106,17 +107,12 @@ def _solve_challenges(self):
         """Get Responses for challenges from authenticators."""
         cont_resp = []
         dv_resp = []
-        try:
+        logger.info("Attempting to set up challenges.")
+        with error_handler.ErrorHandler(self._cleanup_challenges):
             if self.cont_c:
                 cont_resp = self.cont_auth.perform(self.cont_c)
             if self.dv_c:
                 dv_resp = self.dv_auth.perform(self.dv_c)
-        # This will catch both specific types of errors.
-        except errors.AuthorizationError:
-            logger.critical("Failure in setting up challenges.")
-            logger.info("Attempting to clean up outstanding challenges...")
```

Comment:
I think it would be good to keep those log statements: in the default logging level they will be shown to the user providing useful  feedback